mmv1/products/dataplex/Datascan.yaml

```diff
@@ -0,0 +1,640 @@
+  import_format:
+    ['projects/{{project}}/locations/{{location}}/dataScans/{{name}}', '{{name}}']
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'dataplex_datascan_primary'
+    primary_resource_id: 'example'
+    primary_resource_name:
+      'fmt.Sprintf("tf-test-dataScan%s", context["random_suffix"])'
+    test_env_vars:
+      project_name: :PROJECT_NAME
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'locationId'
+    immutable: true
+    required: true
+    url_param_only: true
+    description: |
+      The location where the data scan should reside.
+properties:
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    output: true
+    description: |
+      The relative resource name of the scan, of the form: projects/{project}/locations/{locationId}/dataScans/{datascan_id}, where project refers to a project_id or project_number and locationId refers to a GCP region.
+  - !ruby/object:Api::Type::String
+    name: 'uid'
+    output: true
+    description: |
+      System generated globally unique ID for the scan. This ID will be different if the scan is deleted and re-created with the same name.
+  - !ruby/object:Api::Type::String
+    name: 'description'
+    description: |
+      Description of the scan.
+  - !ruby/object:Api::Type::String
+    name: 'displayName'
+    description: |
+      User friendly display name.
+  - !ruby/object:Api::Type::KeyValuePairs
+    name: 'labels'
+    description: |
+      User-defined labels for the scan. A list of key->value pairs.
+  - !ruby/object:Api::Type::Enum
+    name: 'state'
+    output: true
+    description: |
+      Current state of the DataScan.
+    values:
+      - :STATE_UNSPECIFIED
+      - :ACTIVE
+      - :CREATING
+      - :DELETING
+      - :ACTION_REQUIRED
+  - !ruby/object:Api::Type::String
+    name: 'createTime'
+    output: true
+    description: |
+      The time when the scan was created.
+  - !ruby/object:Api::Type::String
+    name: 'updateTime'
+    output: true
+    description: |
+      The time when the scan was last updated.
+  - !ruby/object:Api::Type::NestedObject
+    name: 'data'
+    required: true
+    immutable: true
+    description: |
+      The data source for DataScan.
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'entity'
+        immutable: true
+        description: |
+          The Dataplex entity that represents the data source(e.g. BigQuery table) for Datascan.
+  - !ruby/object:Api::Type::NestedObject
+    name: 'executionSpec'
+    description: |
+      DataScan execution settings.
+    properties:
+      - !ruby/object:Api::Type::NestedObject
+        name: 'trigger'
+        description: |
+          Spec related to how often and when a scan should be triggered. If not specified, the default is OnDemand, which means the scan will not run until the user calls dataScans.run API.
+        properties:
+          - !ruby/object:Api::Type::NestedObject
+            name: 'onDemand'
```

Comment:
Does [onDemand](https://cloud.google.com/dataplex/docs/reference/rest/v1/projects.locations.dataScans#OnDemand) have a nested `cron` string as well?

Reply:
onDemand (and several RuleType fields below are currently an empty message with no properties;
The idea is to keep it as a message so that we can add fields to them later if needed.

User still needs to specify that they want onDemand instead of Scheduled though.

----
For empty messages in the API, should we just have NestedObject with no property in the YAML?